manifests/server/root_password.pp

```diff
@@ -2,12 +2,27 @@
 class mysql::server::root_password {
 
   $options = $mysql::server::options
+  $secret_file = $mysql::server::install_secret_file
+
+  # New installations of MySQL will configure a default random password for the root user
+  # with an expiration. No actions can be performed until this password is changed. The
+  # below exec will remove this default password. If the user has supplied a root
+  # password it will be set further down with the mysql_user resource.
+  $ch_pass_command = join([
+    "/usr/bin/mysqladmin -u root --password=\$(awk '/password/{print \$NF}' ${secret_file}) password ''",
```

Comment:
Thanks for the feedback. As I said, I didn't expect this to get merged. This was a quick attempt at addressing the problem. I was half expecting you to ask that I replace the exec with a type/provider.

I will change the expression used to retrieve the password so its clearer, I will remove the hardcoded paths and add the path parameter, and I will add tests to cover this.